comparisontool/static/comparisontool/js/comparison-tool.js

```diff
@@ -2270,58 +2282,65 @@ var CFPBComparisonTool = (function() {
 
             // toggle save drawer
             $("#save-and-share").click( function( event, native ) {
-                if ( global.worksheet_id == "none") {
-                    getWorksheetID();
-                }
-                var posturl = "api/worksheet/" + global.worksheet_id + ".json";
-                
-                // put schoolData into a nice JSON object
-                var json_schools = new Object;
-                $.each(schools, function(key, val) {
-                    var data = val.schoolData;
-                    json_schools[key] = data;                        
-                });
-                json_schools = JSON.stringify( json_schools );
-                var request = $.ajax({
-                    type: "POST",
-                    url: posturl,
-                    dataType: "JSON",
-                    data: json_schools
-                });
-                request.done( function ( result ) {
+                $.when( getWorksheetID() )
+                  .done( function() {
+                    var posturl = "api/worksheet/" + global.worksheet_id + ".json";
+
+                    // put schoolData into a nice JSON object
+                    var json_schools = new Object;
+                    $.each(schools, function(key, val) {
```

Comment:
Minor: inconsistent spacing within parens throughout this file.

Reply:
This file is full of bad linting errors. I hope to fix that, refactor a lot of it, and add tests in another PR.